### ISSUE TITLE
Results are not unique when matches in more than one field

### ISSUE THREAD

**Issue by manuelportela** [nextapps-de/flexsearch#38]
I expected to get matching documents to be unique within result. What is the angle for repeating these?

Example:
```javascript
const f = new FlexSearch({
	doc: {
		id: 'id',
		field: ['field1', 'field2']
	}
})

const docs = [
	{id: 1, field1: 'phrase', field2: 'phrase'}
]

f.add(docs)
console.log(f.search('phrase'))
// Result = [{id: 1, field1: "phrase", field2: "phrase"} 1: {id: 1, field1: "phrase", field2: "phrase"}]
```

**Comment by ts-thomas** [comment]
Simply pass field in a custom search:

```js
console.log(f.search('phrase', {field: 'field1'}));
console.log(f.search('phrase', {field: 'field2'}));
```

**Comment by manuelportela** [comment]
That works indeed, but in my case I want all results that have a match somewhere but I only care for the matching document to appear once in the result. I can manage by filtering afterwards, but this could clash when using a limit search.
Why do you return the resulting docs more then once?

**Comment by ts-thomas** [comment]
This issue refers to: https://github.com/nextapps-de/flexsearch/milestone/7
The merging of results from different fields provides this issue actually and should be fixed with this milestone.

**Comment by ts-thomas** [comment]
This issue is fixed (v0.5.2).
